### PR TITLE
Improve runtime

### PR DESCRIPTION
**Pull request by sambenfredj**
- Expose max workers parameter to cli
- update triqler package with the fix for  [issue #13](https://github.com/statisticalbiotechnology/triqler/issues/13) (calculating PEPs may be extremely slow  for some runs)
- make grid search run with multi-threading

image

### CODE REVIEW

File: mokapot/config.py [hunk 1]
```diff
@@ -65,6 +65,14 @@ def _parser():
         ),
     )
 
+    parser.add_argument(
+        "-w",
+        "--max_workers",
+        default=1,
+        type=int,
+        help="The number of processes to use for model training."
```
Comment (wfondrie) [inline mokapot/config.py:73]: This line needs a trailing comma to pass our lint tests.

File: setup.cfg [hunk 2]
```diff
@@ -25,9 +25,9 @@ install_requires =
     pandas>=1.0.3
     scikit-learn>=0.22.1
     numba>=0.48.0
-    triqler>=0.3.0
     matplotlib>=3.1.3
     lxml>=4.6.2
+    triqler @ git+ssh://git@github.com/statisticalbiotechnology/triqler.git
```
Comment (wfondrie) [inline setup.cfg:30]: I don't like having a non-PyPI dependency here, although it seems to make a large difference. I'll look into why this update isn't on PyPI yet.

**Review by wfondrie — CHANGES_REQUESTED** [review]
Hi @sambenfredj - thanks for the great PR! 🎉 

It looks like the code currently fails our lint tests. One way to fix this is to use [Black](https://black.readthedocs.io/en/stable/) as a pre-commit hook. To do this, first install pre-commit:
```
# With pip:
pip install pre-commit

# With conda:
conda install -c conda-forge pre-commit 
```

Once you have pre-commit installed, you can set up the hook with:
```
pre-commit install
```
Now black will run automatically with each new commit.

Lastly, we should add a test that incorporates the new `max_workers` CLI argument. I think it may be too difficult to test whether the argument is truly working, so my suggestion is merely to add it to our basic CLI system test here:
https://github.com/wfondrie/mokapot/blob/bed4946e94690c6fe8ca13299b58a5cfb7eda285/tests/system_tests/test_cli.py#L45-L72

Please don't hesitate to reach out if you have any questions!

**Comment by wfondrie** [comment]
It looks like tests are failing currently, because our GitHub Actions workflows are not setup to clone repos with ssh; hence, it fails to install mokapot with the changed Triqler depedency. 

I've reached out the Triqler folks to get an ETA on when the recent patch will be in a release. If it is relatively soon, we should revert the dependency change in `setup.cfg` and instead bump the minimum version to the new release once it is live.

**Comment by gessulat** [comment]
That totally makes sense. We heard that @MatthewThe is on vacation currently and were just excited to get this out as quickly as possible ;)
Thank you for the super quick response!

**Comment by MatthewThe** [comment]
I updated Triqler on PyPI (v0.6.2), let me know if you run into any other issues.

**Comment by wfondrie** [comment]
Thanks @MatthewThe!

@sambenfredj and @gessulat - if you change the triqler dependency to  `triqler>=0.6.2`, the test should be able to run. Once we confirm all of the tests are passing, it'll be ready to merge.

**Comment by sambenfredj** [comment]
Thanks @wfondrie ! I just updated the triqler to the latest release.

**Comment by tkschmidt** [comment]
Hey @wfondrie,
I'm taking over for @sambenfredj  as she is on vacation. I'm not really firm with the test pipelines in github. Can we trigger it again just to see if it resolves it. As far as I know it worked on our local systems (bad excuse ;))

**Comment by wfondrie** [comment]
I'm a bit stumped as to why the Ubuntu test hangs for hours. I'll try kicking it off again and see if it completes this time. 

Either way, I'll merge this into master. However, I'm going to hold off creating a new release if it doesn't work, so that I can troubleshoot it a little more.

**Comment by wfondrie** [comment]
mokapot v0.8.0 is now on PyPI 🎉 

It should be up on bioconda as well within the next few days.

**Comment by wfondrie** [comment]
Hi @sambenfredj, @tkschmidt, and @gessulat - I'm planning to tweet about recent updates to things soon and I wanted to mention your contribution here. I can mention you each individually (@sambenfredj - do you have a twitter handle?), MSAID, or both - whatever your preference 😄. 

It's also fine if you don't want to be mentioned - please just let me know.

**Comment by sambenfredj** [comment]
Thank you @wfondrie, I am glad to be mentioned. You can find me under the twitter handle **SamiaFredj**.

**Comment by gessulat** [comment]
Hi @wfondrie,
thank you so much, I would be happy about being mentioned but it was really all @sambenfredj who did the all the work. So if it's to much names for a tweet just leave me out ;)

**Pull request merged**